Sources/NIOSSL/NIOSSLHandler+Configuration.swift

```diff
@@ -0,0 +1,38 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+extension NIOSSLHandler {
+    /// Configuration for a specific instance of ``NIOSSLHandler``, either client or server.
+    ///
+    /// This type is distinct from ``TLSConfiguration`` because it does not contain settings that
+    /// apply to TLS itself. Instead, this configuration manages how the ``NIOSSLHandler`` itself
+    /// operates.
+    public struct Configuration: Hashable, Sendable {
```

Comment:
I get a bit nervous adding `Hashable` to configuration objects in case we want to add non-`Hashable` configuration later (callbacks etc.)

Reply:
Ah, I feel the other way around: having them be `Hashable` up front helps you consider what a better strategy would be than callbacks in config objects.

Reply:
Fair!

Reply:
I am kinda leaning towards George lately. I have been bitten by making `Configuration` objects `Hashable` lately. So I am also leaning towards not making them `Hashable`. I am curious when people actually want to check them for equality or store them in a dict.

Reply:
In this particular case I guess I get it, but `TLSConfiguration` is the standard counter-example. It's used to build an `SSLContext`, which is an expensive object, and so it is frequently used as a cache key.